### PR TITLE
KEYCLOAK-13357 Upgrade MariaDB driver

### PR DESCRIPTION
**Pull request by hmlnarik**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Review by mhajas — CHANGES_REQUESTED** [review]
We should remove also tests: https://github.com/jboss-container-images/redhat-sso-7-openshift-image/blob/sso73-dev/tests/arq/src/test/java/org/jboss/test/arquillian/ce/sso/Sso72ServerMysqlTest.java

**Review by mhajas — APPROVED** [review]
LGTM, thanks @hmlnarik .

File: image.yaml [hunk 1]
```diff
@@ -103,8 +103,8 @@ modules:
           - name: sso.config.launch.setup.74
           - name: sso.db.drivers
             version: '1.0'
-          - name: sso.datasource.overrides
-            version: '1.0'
+          # - name: sso.datasource.overrides
+          #   version: '1.0'
```
Comment (iankko) [inline image.yaml]: @hmlnarik Do we want to remove this module altogether or intentionally keeping it for any future re-use?

**Comment by iankko** [comment]
Could you also drop:
* `docs/templates/sso74-x509-mysql-persistent.adoc`
* `docs/templates/sso74-mysql.adoc`
* `docs/templates/sso74-mysql-persistent.adoc`
* `docs/templates/openj9/sso74-openj9-x509-mysql-persistent.adoc`
* `docs/templates/openj9/sso74-openj9-mysql.adoc`
* `docs/templates/openj9/sso74-openj9-mysql-persistent.adoc`

Comment out / remove the following two lines from `modules/sso/config/launch/setup/74/added/launch/datasource.sh`:
```
 grep -r -A1 'KEYCLOAK-10694 - Replace' modules/sso/config/launch/setup/74/added/launch/datasource.sh
  # KEYCLOAK-10694 - Replace any request for MySQL DB driver with a request for the MariaDB one in RHEL-8 UBI minimal
  driver=${driver//mysql/mariadb}
```

Could you also create the corresponding (dropping mention about the MySQL templates) PR for the image documentation?

The rest LGTM

File: image.yaml (hunk 1 above)
Comment (hmlnarik) [reply image.yaml]: Thanks @iankko. The overrides were there to account for mariaDB, so this module might not be necessary anymore. I need to check whether the without the overrides, the image behaves correctly. If so, we can remove it completely.

**Review by iankko — APPROVED** [review]
This version LGTM, thanks Hynek!

**Comment by hmlnarik** [comment]
Docs PR: https://github.com/keycloak/keycloak-documentation/pull/890

**Comment by iankko** [comment]
Confirmed the cli extension works OK. Having a `sso-extensions.cli` in the form e.g. of:

```
[iankko@localhost custom_driver]$ cat sso-extensions.cli 
batch

set DB_DRIVER_NAME=mariadb
set DB_USERNAME=keycloak
set DB_PASSWORD=password
set DB_DRIVER=org.mariadb.jdbc.Driver
set DB_XA_DRIVER=org.mariadb.jdbc.MariaDbDataSource
set DB_JDBC_URL=jdbc:mariadb://sso-mysql.iankko-custom-driver-multiline-demo.svc.cluster.local/keycloak
set DB_EAP_MODULE=org.mariadb

set FILE=/opt/eap/extensions/jdbc-driver.jar

module add --name=$DB_EAP_MODULE --resources=$FILE --dependencies=javax.api,javax.resource.api
/subsystem=datasources/jdbc-driver=$DB_DRIVER_NAME:add( \
  driver-name=$DB_DRIVER_NAME, \
  driver-module-name=$DB_EAP_MODULE, \
  driver-class-name=$DB_DRIVER, \
  driver-xa-datasource-class-name=$DB_XA_DRIVER \
)
/subsystem=datasources/data-source=KeycloakDS:remove()
/subsystem=datasources/data-source=KeycloakDS:add( \
  jndi-name=java:jboss/datasources/KeycloakDS, \
  enabled=true, \
  use-java-context=true, \
  connection-url=$DB_JDBC_URL, \
  driver-name=$DB_DRIVER_NAME, \
  user-name=$DB_USERNAME, \
  password=$DB_PASSWORD \
)

run-batch
```
It's run / executed properly:


7 4_custom_cli_extension

**Review by iankko — APPROVED** [review]
LGTM

**Pull request merged**